File: Sum of Proper Divisors/SumOfProperDivisors.py
import math

def divsum(n):
        if n == 1:
                return 0
        res = 0
        i = 2
        while( i <= math.sqrt(n)):
                if(n%i == 0):
                        if(i == (n/i)):
                                res += i
                        else:
                                res += (i + n/i)
                i += 1
        return (int)(res+1)


n = int(input("Enter the nth value to find :"))
print(n,"th number sum of all divisors is : ",divsum(n))

File: Sum of Proper Divisors/SumOfProperDivisors.java
import java.util.Scanner;

public class SumOfProperDivisors {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.print("Enter the nth value to find : ");
        int n = scan.nextInt();
        System.out.println(n + "th value in sum of all divisors is :" + seq(n));
    }

    private static int seq(int n) {
        int res = 0;
        int i = 2;
        while(i <= Math.sqrt(n)){
            if(n % i == 0){
                res = (i == (n/i)) ? res+i : res + (i + (n/i));
            }
            i += 1;
        }
        return (res+1);
    }
}
